dotnet/src/SemanticKernel.Core/Orchestration/KernelExecutionContext.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Services;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel execution context implementation
+/// This will live during the execution of functions.
+/// </summary>
+internal sealed class KernelExecutionContext : IKernelExecutionContext, IDisposable
+{
+    private Kernel _kernel;
+
+    /// <inheritdoc/>
+    public ILoggerFactory LoggerFactory => this._kernel.LoggerFactory;
+
+    /// <inheritdoc/>
+    public IReadOnlyFunctionCollection Functions => this._kernel.Functions;
+
+    internal KernelExecutionContext(Kernel kernel,
+        IReadOnlyFunctionCollection functionCollection,
+        IAIServiceProvider aiServiceProvider)
+    {
+        this._kernel = new Kernel(
+            new FunctionCollection(functionCollection),
+            aiServiceProvider,
+            kernel.PromptTemplateEngine,
+            kernel.Memory,
+            kernel.HttpHandlerFactory,
+            kernel.LoggerFactory);
+    }
+
+    /// <inheritdoc/>
+    public Task<SKContext> RunAsync(ContextVariables variables, ISKFunction[] pipeline, CancellationToken cancellationToken = default)
```

Comment:
I would much rather have only this version of RunAsync in this interface (no pipelines):
    
public Task<SKContext> RunAsync(ISKFunction skFunction,
    ContextVariables? variables = null,
    CancellationToken cancellationToken = default)

Hopefully that alone is enough (I don't think we need to add the other extension overloads.) If we do, please only add strictly the ones we need today to unblock what's in the repo.

Reply:
Agreed, overloads removed, using the suggested one.